src/api/utils/TodoContext/TodoContext.tsx

```diff
@@ -0,0 +1,275 @@
+import React, { useEffect, useState } from 'react';
+
+import { UserWarning } from '../../../UserWarning';
+import {
+  createTodo, getTodos, deleteTodo, updateTodoItem,
+} from '../../todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  children: React.ReactNode;
+};
+
+const USER_ID = 12157;
+
+export enum FilterStatus {
+  All = 'All',
+  Active = 'Active',
+  Completed = 'Completed',
+}
+
+type ContextType = {
+  todos: Todo[];
+  setTodos: (todoArr: Todo[]) => void;
+  query: string;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  isDisabled: boolean,
+  setIsDisabled: React.Dispatch<React.SetStateAction<boolean>>;
+  filt: FilterStatus;
+  setFilt: React.Dispatch<React.SetStateAction<FilterStatus>>;
+  setError: React.Dispatch<React.SetStateAction<string | null>>
+  error: string | null;
+  tempTodo: Todo | null;
+  onSubmit: (title: string) => void;
+  handleError: (errorMessage: string) => void;
+  filteredTodos: Todo[];
+  TodoDeleteButton: (item: number) => void;
+  clearCompleted: () => void;
+  handleUpdate: (todo: Todo) => void;
+  updateTodo: (todo: Todo) => void;
+  completeAll: () => void;
+  loaderTodoId: number[] | null;
+  setLoaderTodoId: (number: number[] | null) => void;
+};
+
+export const TodosContext = React.createContext<ContextType>({
+  todos: [],
+  setTodos: () => {},
+  query: '',
+  setQuery: () => {},
+  isDisabled: false,
+  setIsDisabled: () => {},
+  filt: FilterStatus.All,
+  setFilt: () => {},
+  setError: () => {},
+  error: '' || null,
+  tempTodo: null,
+  onSubmit: () => {},
+  handleError: () => {},
+  filteredTodos: [],
+  TodoDeleteButton: () => {},
+  clearCompleted: () => {},
+  handleUpdate: () => {},
+  updateTodo: () => {},
+  completeAll: () => {},
+  loaderTodoId: [] || null,
+  setLoaderTodoId: () => {},
+});
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filt, setFilt] = useState(FilterStatus.All);
+  const [error, setError] = useState<string | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [loaderTodoId, setLoaderTodoId] = useState<number[] | null>(null);
```

Comment:
If it will be array use plurals for naming
```suggestion
  const [loaderTodoIds, setLoaderTodoIds] = useState<number[] | null>(null);
```